bootstrap/kubeadm/api/v1beta1/kubeadm_types.go

```diff
@@ -512,6 +513,29 @@ type BootstrapTokenDiscovery struct {
 type FileDiscovery struct {
 	// KubeConfigPath is used to specify the actual file path or URL to the kubeconfig file from which to load cluster information
 	KubeConfigPath string `json:"kubeConfigPath"`
+
+	// KubeConfig is used (optionally) to generate a KubeConfig based on the Cluster's information.
+	// The kubeconfig is generated with a server and context matching the Cluster's name,
+	// Host address (server field) information is automatically populated based on the Cluster's ControlPlaneEndpoint.
+	// Certificate Authority (certificate-authority-data field) is gathered from the cluster's CA secret.
+	// +optional
+	KubeConfig *FileDiscoveryKubeConfig `json:"kubeConfig,omitempty"`
+}
+
+// FileDiscoveryKubeConfig contains elements describing how to generate the kubeconfig for bootstrapping.
+type FileDiscoveryKubeConfig struct {
+	// Cluster contains information about how to communicate with the kubernetes cluster.
+	//
+	// By default the following fields are automatically populated:
+	// - Name with the Cluster's name.
+	// - Server with the Cluster's ControlPlaneEndpoint.
+	// - CertificateAuthorityData with the Cluster's CA certificate.
+	// +optional
+	Cluster *clientcmdv1.Cluster `json:"cluster,omitempty"`
+
+	// User contains information that describes identity information.
+	// This is use to tell the kubernetes cluster who you are.
+	User clientcmdv1.AuthInfo `json:"user"`
```

Comment:
this diverges from the upstream.
are there other cases of that in the forked CAPI kubeadm types?

Reply:
We're just ignoring these fields when they convert down. While it's diverging the types (which I don't think it's a big problem), we're converting these to v1beta1/3/4 on the kubeadm side and change fields as needed already

Reply:
it might be better to add these fields to the CAPI types instead.
EOD it doesn't matter much IMO, but i think the kubeadm forks should be left vanilla to avoid reader confusion.

Reply:
These are the CAPI types equivalent, the forks aren't being touched 😄